core/controllers/editor.py

```diff
@@ -101,9 +101,30 @@ def _require_valid_version(version_from_payload, exploration_version):
             % (exploration_version, version_from_payload))
 
 
+class EditorLogoutHandler(base.LogoutPage):
+    """Handles logout from editor page."""
+
+    def get(self):
+        """Checks if exploration is published and redirects accordingly."""
+
+        return_url = str(self.request.get('return_url'))
+        if return_url:
+            exploration_id = return_url.split('/')[-1]
+            exp_summary = exp_services.get_exploration_summary_by_id(
```

Comment:
Need to handle the case where the exploration_id is invalid (users can hypothetically enter anything they like in the URL bar) and add a backend test for it.

Reply:
yes, will do it.

Reply:
i have handled that using '@acl_decorators.can_play_exploration'.